### PR TITLE
fix(integration-tests): enable auth by default

### PR DESCRIPTION
**Pull request by fruch**
for covering related auth in alternator tests
we want to enable the integration tests to work
by default with auth enforced.

it flushed out, that we need to fix latte support
and that cql-stress version we have doesn't support auth

### Testing

- [ ]

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: unit_tests/test_latte_thread.py [hunk 1]
```diff
@@ -106,8 +104,7 @@ def check_metrics():
 def test_04_latte_run_client_encrypt(request, docker_scylla, params):
     params['client_encrypt'] = True
 
-    loader_set = LocalLoaderSetDummy()
-    loader_set.params = params
+    loader_set = LocalLoaderSetDummy(params=params)
 
     cmd = ("latte run -d 10s docker/latte/workloads/workload.rn")
```
Comment (vponomaryov) [inline unit_tests/test_latte_thread.py:109]: add `--generate-report` param to the latte command to fix the breakage.

File: unit_tests/test_latte_thread.py [hunk 2]
```diff
@@ -67,8 +66,7 @@ def cleanup_thread():
 
 
 def test_03_latte_run(request, docker_scylla, prom_address, params):
-    loader_set = LocalLoaderSetDummy()
-    loader_set.params = params
+    loader_set = LocalLoaderSetDummy(params=params)
 
     cmd = ("latte run --function run -d 10s docker/latte/workloads/workload.rn")
```
Comment (vponomaryov) [inline unit_tests/test_latte_thread.py:71]: add `--generate-report` param to the latte command to fix the breakage.

File: unit_tests/conftest.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ def db_alternator_up():
             logging.error("Error checking for scylla up normal: %s", details)
             return False
 
-    scylla.remoter.run('apt-get update')
+    # scylla.remoter.run('apt-get update')
```
Comment (vponomaryov) [inline unit_tests/conftest.py:105]: Is it debug left-over?

File: unit_tests/test_latte_thread.py (hunk 2 above)
Comment (vponomaryov) [reply unit_tests/test_latte_thread.py:71]: Added separate PR: https://github.com/scylladb/scylla-cluster-tests/pull/7586

File: unit_tests/test_latte_thread.py (hunk 1 above)
Comment (vponomaryov) [reply unit_tests/test_latte_thread.py:109]: Added separate PR: https://github.com/scylladb/scylla-cluster-tests/pull/7586

File: unit_tests/conftest.py (hunk 3 above)
Comment (fruch) [reply unit_tests/conftest.py:105]: I think this is not needed anymore, I need to put it into it's own commit

Comment (fruch) [reply unit_tests/conftest.py]: move it to it's own commit [3cf3c4c](https://github.com/scylladb/scylla-cluster-tests/pull/7579/commits/3cf3c4cf296d6bfbb40c9ab37441b1702cf19d7a)

File: docker/cql-stress-cassandra-stress/image [hunk 4]
```diff
@@ -1 +1 @@
-scylladb/hydra-loaders:cql-stress-cassandra-stress-20240119
+docker.io/scylladb/hydra-loaders:cql-stress-cassandra-stress-20240606
```
Comment (soyacz) [inline docker/cql-stress-cassandra-stress/image:1]: why docker.io here?

**Comment by soyacz** [comment]
integration tests are failing

**Comment by fruch** [comment]
> integration tests are failing

Yeah, since we are not attending to it a lot, and I've update the scylla version used

There's multiple things to attend to

We might consider star using it with latest scylla, and maybe make some of those mandatory

**Review by vponomaryov — CHANGES_REQUESTED** [review]
Overall LGTM,

Need to remove `latte broken` mentioning in the commit message and PR description because it was fixed some time ago.

Also, need to rebase this change.

**Comment by fruch** [comment]
> Overall LGTM,
> 
> Need to remove `latte broken` mentioning in the commit message and PR description because it was fixed some time ago.
> 
> Also, need to rebase this change.


rebased

and the latte broken, was about SCT latte code missing auth support.

**Comment by fruch** [comment]
And there are a few more fixes needed to the integration tests, cause of updating to newer scylla image

**Comment by vponomaryov** [comment]
@fruch unit tests fail the following way:

```
23:11:41  ==================================== ERRORS ====================================
23:11:41  _______________ ERROR collecting test_alternator_streams_kcl.py ________________
23:11:41  ImportError while importing test module '/tmp/jenkins/workspace/can_scylla-cluster-tests_PR-7579/unit_tests/test_alternator_streams_kcl.py'.
23:11:41  Hint: make sure your test modules/packages have valid Python names.
23:11:41  Traceback:
23:11:41  /usr/local/lib/python3.10/importlib/__init__.py:126: in import_module
23:11:41      return _bootstrap._gcd_import(name[level:], package, level)
23:11:41  unit_tests/test_alternator_streams_kcl.py:21: in <module>
23:11:41      from unit_tests.lib.alternator_utils import TEST_PARAMS
23:11:41  E   ImportError: cannot import name 'TEST_PARAMS' from 'unit_tests.lib.alternator_utils' (/tmp/jenkins/workspace/can_scylla-cluster-tests_PR-7579/unit_tests/lib/alternator_utils.py)
```

everything else lgtm.

**Comment by fruch** [comment]
> @fruch unit tests fail the following way:
> 
> ```
> 23:11:41  ==================================== ERRORS ====================================
> 23:11:41  _______________ ERROR collecting test_alternator_streams_kcl.py ________________
> 23:11:41  ImportError while importing test module '/tmp/jenkins/workspace/can_scylla-cluster-tests_PR-7579/unit_tests/test_alternator_streams_kcl.py'.
> 23:11:41  Hint: make sure your test modules/packages have valid Python names.
> 23:11:41  Traceback:
> 23:11:41  /usr/local/lib/python3.10/importlib/__init__.py:126: in import_module
> 23:11:41      return _bootstrap._gcd_import(name[level:], package, level)
> 23:11:41  unit_tests/test_alternator_streams_kcl.py:21: in <module>
> 23:11:41      from unit_tests.lib.alternator_utils import TEST_PARAMS
> 23:11:41  E   ImportError: cannot import name 'TEST_PARAMS' from 'unit_tests.lib.alternator_utils' (/tmp/jenkins/workspace/can_scylla-cluster-tests_PR-7579/unit_tests/lib/alternator_utils.py)
> ```
> 
> everything else lgtm.

Fixed it, finally all integration suite is passing

**Pull request merged**